File: BOJ/10000 ~ 11000/P10811.java
// https://www.acmicpc.net/problem/10811

import java.util.*;
import java.io.*;

public class P10811 {
    public static void main(String[] args) throws IOException {
        InputStreamReader isr = new InputStreamReader(System.in);
        BufferedReader br = new BufferedReader(isr);
        StringTokenizer st = new StringTokenizer(br.readLine());

        // N M
        int n = Integer.parseInt(st.nextToken());
        int m = Integer.parseInt(st.nextToken());

        // N개 바구니 만들기
        int[] baskets = new int[n+1];

        // init basket
        for (int i=1; i<=n; i++) {
            baskets[i] = i;
        }

        for (int i=1; i<=m; i++) {
            st = new StringTokenizer(br.readLine());

            int iBasket = Integer.parseInt(st.nextToken());
            int jBasket = Integer.parseInt(st.nextToken());
            int[] temp = new int[jBasket - iBasket + 1];
            for (int k = iBasket; k <= jBasket; k++) {
                temp[k - iBasket] = baskets[k];
            }

            for(int k = iBasket; k <= jBasket; k++) {
                baskets[k] = temp[jBasket - k];
            }
        }

        for (int i=1; i<=n; i++) {
            if (i != n) {
                System.out.print(baskets[i] + " ");
            } else {
                System.out.print(baskets[i]);
            }
        }
    }
}

File: BOJ/10000 ~ 11000/P10811.kt
import java.io.*

// https://www.acmicpc.net/problem/10811

fun main(args: Array<String>) = with(
    BufferedReader(InputStreamReader(System.`in`))
) {
    // n: 바구니의 개수, m: 바꿔치기 횟수
    val (n, m) = readLine().split(" ").map { it.toInt() }
    val baskets: Array<Int> = Array(n+1) { 0 }

    // init baskets
    for (i in 1 .. n) {
        baskets[i] = i
    }

    // M번 위치를 바꾼다.
    for (i in 0 until m) {
        val (a, b) = readLine().split(" ").map { it.toInt() }
        val temp = baskets.copyOf().slice(IntRange(a, b))

        for (k in a .. b) {
            baskets[k] = temp[b - k]
        }
    }

    for (idx in baskets.indices) {
        if (idx == 0) continue
        if (idx == n) {
            print("${baskets[idx]}")
        } else {
            print("${baskets[idx]} ")
        }
    }
}
